bolt/lib/Core/DebugData.cpp

```diff
@@ -906,6 +906,8 @@ void DebugStrOffsetsWriter::finalizeSection(DWARFUnit &Unit,
   }
 
   StrOffsetSectionWasModified = false;
+  assert(DebugStrOffsetFinalized.insert(Unit.getOffset()).second == true &&
```

Comment:
I think you can just do:
DebugStrOffsetFinalized.insert(Unit.getOffset()).second
maybe add "for a CU" to error message.
Also add assert in updateAddressMap to check we don't call it on already finalized CU.